src/components/FeedbackDialog.tsx

```diff
@@ -0,0 +1,379 @@
+import React, { useState, useCallback, useMemo } from "react";
+import {
+    Button,
+    Checkbox,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogTitle,
+    FormControlLabel,
+    Link,
+    TextField,
+    styled,
+    Box,
+    Tooltip,
+    Divider,
+} from "@material-ui/core";
+import { white, indigo500 } from "material-ui/styles/colors";
+import { useBooleanState } from "../hooks/useBoolean";
+import i18n from "../locales";
+import { HelpOutline } from "@material-ui/icons";
+import { Fields, LayoutOptions } from "../domain/entities/Feedback";
+import { screenshot, checkIfBrowserSupported } from "../data/screenshot";
+import { ClickUp } from "../data/clickup";
+import { ClickUpOptions } from "../domain/entities/ClickUp";
+
+interface FeedbackDialogProps {
+    open: boolean;
+    options?: LayoutOptions;
+    clickUp?: ClickUpOptions;
+    username?: string;
+    onClose: () => void;
+    onSend: (content: string) => void;
+}
+
+export const FeedbackDialog: React.FC<FeedbackDialogProps> = React.memo(
+    ({ open, options, clickUp, onClose, onSend, username }) => {
+        const AgreementLabel: React.ReactNode = useMemo(getAgreementLabel, []);
+        const [includeScreenshot, { toggle: toggleScreenshot }] = useBooleanState(false);
+        const [includeContact, { toggle: toggleContact }] = useBooleanState(false);
+        const [acceptAgreement, { toggle: toggleAgreement }] = useBooleanState(false);
+        const [inputs, setInputs] = useState<Inputs>({
+            title: { value: "", isValid: true },
+            description: {
+                value: options?.descriptionTemplate ?? "",
+                isValid: true,
+            },
+            name: { value: "", isValid: true },
+            email: { value: "", isValid: true },
+        });
+
+        const values: Fields = React.useMemo(
+            () => ({
+                title: inputs.title.value,
+                description: inputs.description.value,
+                contact: includeContact
+                    ? { name: inputs.name.value, email: inputs.email.value }
+                    : undefined,
+            }),
+            [inputs, includeContact]
+        );
+
+        const validate = React.useCallback(() => {
```

Comment:
Much of all of this validation belongs to entities.

Reply:
I'm sorry but I don't understand what you meaning.